bukkit/src/main/java/me/lucko/luckperms/bukkit/migration/MigrationPermissionsEx.java

```diff
@@ -99,6 +104,27 @@ public CommandResult execute(LuckPermsPlugin plugin, Sender sender, Object o, Li
             // migrate data
             migrateEntity(group, lpGroup, groupWeight);
 
+            // migrate ladder/track
+            if (group.isRanked()) {
+                String rankLadder = group.getRankLadder();
+                Track track;
+                if ((track = plugin.getTrackManager().getIfLoaded(rankLadder)) == null) {
+                    plugin.getStorage().createAndLoadTrack(rankLadder, CreationCause.INTERNAL).join();
+                    track = plugin.getTrackManager().getIfLoaded(rankLadder);
+                }
+                track.getGroups().forEach(trackGroupName -> plugin.getStorage().loadGroup(trackGroupName).join());
+                List<Group> trackGroups = track.getGroups().stream().map(trackGroupName -> plugin.getGroupManager().getIfLoaded(trackGroupName)).sorted(groupWeightSorter).collect(Collectors.toList());
+                int position = 0;
+                for (Group trackGroup : trackGroups) {
+                    if (lpGroup.getWeight().orElse(0) > trackGroup.getWeight().orElse(0)) {
+                        position++;
+                    }
+                }
+                if (!track.containsGroup(lpGroup)) {
+                    track.insertGroup(lpGroup, position);
+                    plugin.getStorage().saveTrack(track);
```

Comment:
Save at the end instead of after each iteration - and then use `#join` on the future.